scalus-core/shared/src/main/scala/scalus/compiler/sir/lowering/LoweredValue.scala

```diff
@@ -1319,6 +1319,22 @@ object LoweredValue {
         var printedIdentifiers: Set[String] = Set.empty
     )
 
+    /** Normalize BuiltinArray[A] to BuiltinArray[Data] since UPLC arrays can only hold Data. This
+      * is needed for type unification to succeed when comparing BuiltinArray[BigInt] with
+      * BuiltinArray[Data].
+      */
+    def normalizeArrayType(tp: SIRType): SIRType = tp match {
```

Comment:
What this mean ?

 SIRType should reflect semantics.   Changing SIRType without reason  is incorrect.  If the Data is representation-specific issue, you should have representation.